### PR TITLE
[tests] update MSBuild performance tests

### PR DESCRIPTION
**Pull request by jonathanpeppers**
It's been a while, since I went through and refactored these tests. After seeing a couple build-time regressions in .NET 9, I think it's overdue.

Removing test cases that likely won't catch an issue:

* `Build_Designer_Change` (less concerned about designer now)
* `Build_CSProj_Change`

`Build_From_Clean_DontIncludeRestore` would likely catch similar issues as these two, anyway.

`Build_XAML_Change_RefAssembly` now uses a `net8.0` class library where `$(ProduceReferenceAssembly)` defaults to `true`.

Removed a 500ms addition, that only made sense when we supported both "classic" Xamarin.Android and .NET 6+ projects.

Update the time thresholds to be a bit shorter, in general.

### CODE REVIEW

File: tests/MSBuildDeviceIntegration/Tests/PerformanceTest.cs [hunk 1]
```diff
@@ -332,7 +272,7 @@ public void Build_XAML_Change (bool produceReferenceAssembly, bool install)
 			var lib = new DotNetStandard {
 				ProjectName = "MyLibrary",
 				Sdk = "Microsoft.NET.Sdk",
-				TargetFramework = "netstandard2.0",
+				TargetFramework = "net8.0", // Vanilla project
```
Comment (dellis1972) [inline tests/MSBuildDeviceIntegration/Tests/PerformanceTest.cs:275]: why this change? netstandard2.0 is still supported? should we test both?

Comment (jonathanpeppers) [reply tests/MSBuildDeviceIntegration/Tests/PerformanceTest.cs:275]: `netstandard2.0` doesn't default to `ProduceReferenceAssembly=true`. This is just an attempt to use something more modern, closer to a MAUI project.

This really just tests the timing of an incremental build of a non-Android project, and fast deploying it.
